lib/init/index.js

```diff
@@ -1,65 +1,152 @@
+const modeTransform = require("./transformations/mode/mode");
+const resolveLoaderTransform = require("./transformations/resolveLoader/resolveLoader");
 
 /**
  *
- * Runs yeoman and runs the transformations based on the object
- * built up from an author/user
+ * Runs the transformations from an object we get from yeoman
  *
- * @param {String} options - An path to the given generator
- * @returns {Function} runTransform - Run transformations based on the finished
- * yeoman instance
+ * @param {Object} webpackProperties - Configuration to transform
+ * @param {String} action - Action to be done on the given ast
+ * @returns {Promise} - A promise that writes each transform, runs prettier
+ * and writes the file
  */
 
-function creator(options) {
-	let env = yeoman.createEnv("webpack", null);
-	const generatorName = options
-		? replaceGeneratorName(path.basename(options[0]))
-		: "webpack-default-generator";
-	if (options) {
-		const WebpackGenerator = class extends Generator {
-			initializing() {
-				options.forEach(path => {
-					return this.composeWith(require.resolve(path));
-				});
-			}
-		};
-		env.registerStub(WebpackGenerator, generatorName);
-	} else {
-		env.registerStub(defaultGenerator, "webpack-default-generator");
-	}
+const transformsObject = {
+	entryTransform,
+	outputTransform,
+	contextTransform,
+	resolveTransform,
+	devtoolTransform,
+	targetTransform,
+	watchTransform,
+	watchOptionsTransform,
+	externalsTransform,
+	nodeTransform,
+	performanceTransform,
+	statsTransform,
+	amdTransform,
+	bailTransform,
+	cacheTransform,
+	profileTransform,
+	moduleTransform,
+	pluginsTransform,
+	topScopeTransform,
+	mergeTransform,
+	devServerTransform,
+	modeTransform,
+	parallelismTransform,
+	recordsInputPathTransform,
+	recordsOutputPathTransform,
+	recordsPathTransform,
+	resolveLoaderTransform
+};
 
-	env.run(generatorName).on("end", _ => {
-		if (generatorName !== "webpack-default-generator") {
-			//HACK / FIXME
-			env = env.options.env;
-			return runTransform(env.configuration);
-		} else {
-			return runTransform(env.getArgument("configuration"));
-		}
+module.exports = function runTransform(webpackProperties, action) {
+	// webpackOptions.name sent to nameTransform if match
+	const webpackConfig = Object.keys(webpackProperties).filter(p => {
+		return p !== "configFile" && p !== "configPath";
 	});
-}
+	webpackConfig.forEach(scaffoldPiece => {
+		const config = webpackProperties[scaffoldPiece];
+		const transformations = Object.keys(transformsObject)
+			.map(k => {
+				const stringVal = k.substr(0, k.indexOf("Transform"));
+				if (Object.keys(config.webpackOptions).length) {
+					if (config.webpackOptions[stringVal]) {
+						return [transformsObject[k], config.webpackOptions[stringVal]];
+					} else {
+						return [transformsObject[k], config[stringVal]];
+					}
+				} else {
+					return [transformsObject[k]];
+				}
+			})
+			.filter(e => e[1]);
```

Comment:
Using `e[1]` wouldn't pass the last case of map `return [transformsObject[k]]`, but it'll filter with a falsy iteration as `e[1]` being `undefined`.
Can we clean the `else` clause of `map`?

Reply:
It's working as intended

Reply:
Yeah, It'll get cleared anyways because of falsy iteration. 👍